core/src/wallet/tezos/TezosLikeAccount2.cpp

```diff
@@ -307,8 +307,33 @@ namespace ledger {
                                             tx->setGasLimit(request.gasLimit);
                                             tx->setStorage(request.storageLimit);
 
-                                            tx->setSender(accountAddress);
-                                            tx->setReceiver(TezosLikeAddress::fromBase58(request.toAddress, currency));
+                                            auto getCurveHelper = [] (const std::string &xpubConfig) -> api::TezosCurve {
+                                                if (xpubConfig == api::TezosConfigurationDefaults::TEZOS_XPUB_CURVE_ED25519) {
+                                                    return api::TezosCurve::ED25519;
+                                                } else if (xpubConfig == api::TezosConfigurationDefaults::TEZOS_XPUB_CURVE_SECP256K1) {
+                                                    return api::TezosCurve::SECP256K1;
+                                                }
+                                                return api::TezosCurve::P256;
+                                            };
+                                            // Get sender's curve first
+                                            // For KT accounts, it is always ED25519
+                                            auto senderCurve = api::TezosConfigurationDefaults::TEZOS_XPUB_CURVE_ED25519;
+                                            if (senderAddress.find("KT1") != 0) {
+                                                senderCurve = self->getKeychain()->getConfiguration()
```

Comment:
Is it valid if `std::find` returns `std::string::npos` ?

Reply:
Here we test that the address prefix is `KT1` to test if it's an originated account